apps/condo/domains/organization/access/Organization.js

```diff
@@ -78,16 +78,44 @@ async function canReadOrganizations (args) {
             deletedAt: null,
         })
 
-        const serviceOrganizationIds = uniq(billingContexts
+        let serviceOrganizationIds = uniq(billingContexts
             .map(({ organization }) => organization )
             .concat(acquiringContexts.map(({ organization }) => organization )))
             .concat(bankIntegrationOrganizationContext.map(({ organization }) => organization))
             .concat(get(accessFilterForB2BAppServiceUser, 'id_in', []) || [])
+
         if (serviceOrganizationIds.length) {
             accessConditions.push({ id_in: serviceOrganizationIds })
         }
     }
-    return { OR: accessConditions }
+
+    const allowedOrganizations = get(user, 'extra.allowedOrganizations')
```

Comment:
looks like, you can add this code to accessFilterForB2BAppServiceUser

Reply:
Already did, but should duplicate it here, because in case of service account for billing integrations related organizations being added here, separately from accessFilterForB2BAppServiceUser